addons/account/models/account_tax.py

```diff
@@ -1184,6 +1182,13 @@ class AccountTaxRepartitionLine(models.Model):
         check_company=True,
         help="The tax set to apply this distribution on refund invoices. Mutually exclusive with invoice_tax_id")
     tax_id = fields.Many2one(comodel_name='account.tax', compute='_compute_tax_id')
+    document_type = fields.Selection(
+        selection=[
+            ('invoice', 'Invoice'),
+            ('refund', 'Refund'),
+        ],
+        compute='_compute_document_type',
```

Comment:
useless rant: I which it would have been implemented like that at the beginning instead of 2 mutually exclusive m2o needing constraints and weird joins everywhere...